txs/txs_builder.go

```diff
@@ -168,7 +169,7 @@ func orderTXs(logger log.Log, pmd *proposalMetadata, realState stateFunc, blockS
 	// we do not want to access database here, as all transactions are already in memory.
 	dbLessCache := &cache{
 		logger:    logger,
-		tp:        &nopTP{},
+		db:        sql.InMemory(),
```

Comment:
i didn't notice this before but this seems somewhat strange, do you mean that db is not used in the dbLessCache?

Reply:
correct. the DB-less cache is used for constructing a block. we fill the cache with all the proposal txs, reuse the logic there to organize by accounts, and spit them out flat for a block.

in the regular cache for mempool, db's main purpose is to get more/better txs from db when a earlier nonce is applied (and may relax the balance constraint). for block construction, we already have all the txs we need. @dshulyak

Reply:
to reduce confusion, i've removed db from cache all together, and pass in db as needed.